homeassistant/helpers/template.py

```diff
@@ -1314,12 +1319,59 @@ def urlencode(value):
     return urllib_urlencode(value).encode("utf-8")
 
 
+def _render_with_context(
+    template_str: str, template: jinja2.Template, **kwargs: Any
+) -> str:
+    """Store template being rendered in a ContextVar to aid error handling."""
+    template_cv.set(template_str)
+    return template.render(**kwargs)
```

Comment:
Should we set it to `None` after the render?

Reply:
I was thinking about that. Is there any scenario in which we could print a stale contextvar?

Reply:
It's mainly for garbage collection purposes.